<?xml version="1.0"?>
<launch>
  <arg name="uri13" default="radio://0/90/2M/E7E7E7E70D" />
  <arg name="frame13" default="/vicon/crazyflie13/crazyflie13" />

  <arg name="uri15" default="radio://0/100/2M/E7E7E7E70F" />
  <arg name="frame15" default="/vicon/crazyflie15/crazyflie15" />

  <arg name="uri18" default="radio://0/80/2M/E7E7E7E712" />
  <arg name="frame18" default="/vicon/crazyflie18/crazyflie18" />

  <arg name="joy_dev" default="/dev/input/js0" />

  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <group ns="crazyflie13">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri13)" />
      <param name="tf_prefix" value="crazyflie13" />
      <param name="enable_logging_imu" value="False" />
      <param name="enable_logging_temperature" value="False" />
      <param name="enable_logging_magnetic_field" value="False" />
      <param name="enable_logging_pressure" value="False" />
      <param name="enable_logging_battery" value="False" />
      <param name="enable_logging_packets" value="False" />
      <rosparam>
        genericLogTopics: ["log1"]
        genericLogTopicFrequencies: [100]
        genericLogTopic_log1_Variables: ["stateEstimate.x", "stateEstimate.y", "stateEstimate.z"]
      </rosparam>
    </node>

    <node name="pose" pkg="crazyflie_demo" type="publish_external_position_vicon.py" output="screen">
      <param name="topic" value="$(arg frame13)" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="baselink_broadcaster" args="0 0 0 0 0 0 1 $(arg frame13) /crazyflie13/base_link 100" />
  </group>

  <group ns="crazyflie15">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri15)" />
      <param name="tf_prefix" value="crazyflie15" />
      <param name="enable_logging_imu" value="False" />
      <param name="enable_logging_temperature" value="False" />
      <param name="enable_logging_magnetic_field" value="False" />
      <param name="enable_logging_pressure" value="False" />
      <param name="enable_logging_battery" value="False" />
      <param name="enable_logging_packets" value="False" />
      <rosparam>
        genericLogTopics: ["log1"]
        genericLogTopicFrequencies: [100]
        genericLogTopic_log1_Variables: ["stateEstimate.x", "stateEstimate.y", "stateEstimate.z"]
      </rosparam>
    </node>

    <node name="pose" pkg="crazyflie_demo" type="publish_external_position_vicon.py" output="screen">
      <param name="topic" value="$(arg frame15)" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="baselink_broadcaster" args="0 0 0 0 0 0 1 $(arg frame15) /crazyflie15/base_link 100" />
  </group>


  <group ns="crazyflie18">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri18)" />
      <param name="tf_prefix" value="crazyflie18" />
      <param name="enable_logging_imu" value="True" />
      <param name="enable_logging_temperature" value="False" />
      <param name="enable_logging_magnetic_field" value="False" />
      <param name="enable_logging_pressure" value="False" />
      <param name="enable_logging_battery" value="True" />
      <param name="enable_logging_packets" value="False" />
      <rosparam>
        genericLogTopics: ["log1"]
        genericLogTopicFrequencies: [100]
        genericLogTopic_log1_Variables: ["stateEstimate.x", "stateEstimate.y", "stateEstimate.z"]
      </rosparam>
    </node>

    <node name="pose" pkg="crazyflie_demo" type="publish_external_position_vicon.py" output="screen">
      <param name="topic" value="$(arg frame18)" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="baselink_broadcaster" args="0 0 0 0 0 0 1 $(arg frame18) /crazyflie18/base_link 100" />
  </group>


  <!-- run vicon bridge -->
  <include file="$(find vicon_bridge)/launch/vicon.launch"/>

  <!--enable joystick-->
  <!--node name="joy" pkg="joy" type="joy_node" output="screen" >
      <param name="dev" value="$(arg joy_dev)" />
  </node-->

  <!--node name="rviz" pkg="rviz" type="rviz" args="-d $/home/ruslan/.rviz/imp_cargo_rviz_config.rviz" required="true" /-->
</launch>
